cartography/models/aws/ec2/subnet_networkinterface.py

```diff
@@ -0,0 +1,87 @@
+from dataclasses import dataclass
+
+from cartography.models.aws.ec2.subnet_instance import EC2SubnetToAWSAccount
+from cartography.models.aws.ec2.subnet_instance import EC2SubnetToEC2Instance
+from cartography.models.core.common import PropertyRef
+from cartography.models.core.nodes import CartographyNodeProperties
+from cartography.models.core.nodes import CartographyNodeSchema
+from cartography.models.core.relationships import CartographyRelProperties
+from cartography.models.core.relationships import CartographyRelSchema
+from cartography.models.core.relationships import LinkDirection
+from cartography.models.core.relationships import make_target_node_matcher
+from cartography.models.core.relationships import OtherRelationships
+from cartography.models.core.relationships import TargetNodeMatcher
+
+
+@dataclass(frozen=True)
+class EC2SubnetNetworkInterfaceNodeProperties(CartographyNodeProperties):
```

Comment:
Since many of the properties of all these classes will be shared. I think it better to create and use base classes.

Reply:
Eh, not a fan. I created a discussion here: https://github.com/lyft/cartography/discussions/1238

Let's continue the conversation there since I think this is something worth fleshing out in detail.